Level 2

1998max7362 opened the pull request:
Верстка проекта

Для инпутов не задавал типы, рег выражения и тд., потому что планирую использовать формик

1998max7362 committed:
Задал структуру роутов

1998max7362 committed:
Убрал лишние дивы

1998max7362 committed:
Расставил гриды

1998max7362 committed:
автоматическое изменение цвета хедера работает

1998max7362 committed:
add fonts

1998max7362 committed:
Header готов

1998max7362 committed:
Footer готов

1998max7362 committed:
Main страница вроде работает

1998max7362 committed:
Добавил страничку movies

1998max7362 committed:
Добавил фильмы

1998max7362 committed:
Movies есть

1998max7362 committed:
saved movies

1998max7362 committed:
добавил профиль

1998max7362 committed:
Регистрация и вход

1998max7362 committed:
404 есть

1998max7362 committed:
вроде готово

1998max7362 committed:
поправил инпут

1998max7362 committed:
анимация работает!

1998max7362 committed:
Ура всё работает

1998max7362 committed:
поправил анимацию

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [ ] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
`Отсутствует состояние кнопки в карточке "активированное сердце»`
`Отсутствует состояние кнопки в карточке "крестик»`

- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [ ] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`ссылки class="link student__portfolio-item"  ведут в пустоту`
`ссылки class="link student__portfolio-item"  должны вести на ваши сторонние проекты`
`ссылки на сторонние страницы должны открываться в новой вкладке`
`class="header__logo" это должна быть ссылка на главную страницу`

- [ ] <!--5.55--> Именование классов сделаны по БЭМ.
`Есть ошибки в нейминге, Вы можете проверить их здесь (https://nglazov.github.io/bem-validator-page/)`

- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [ ] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
`Кнопки в карточки необходимо вставлять через тег button`

- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;x
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:ф
	- [x] <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [ ] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
`Кнопка формы поиска фильмов не имеет состояния наведения`

- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы форм не выделяются, когда на них установлен ФОКУС. Форма поиска фильмов`

- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`Форма поиска фильмов не валидируется. У инпутов можно добавить сво-во required`

- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 63

## Доп. Комментарии:
`Неправильно задан lang`

1998max7362 committed:
Исправил все ошибки за исключением:

 Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
Элементы форм не выделяются, когда на них установлен ФОКУС. Форма поиска фильмов

1998max7362 commented:
Огромное спасибо за столь оперативную проверку!!!

Поправил все ошибки кроме:

- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы форм не выделяются, когда на них установлен ФОКУС. Форма поиска фильмов`

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;x
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:ф
	- [x] <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы форм не выделяются, когда на них установлен ФОКУС. Форма поиска фильмов`

- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`Форма поиска фильмов не валидируется. У инпутов можно добавить сво-во required`

- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 96

## Доп. Комментарии:
